### DISCUSSION TITLE
Socket() in spike prime

### DISCUSSION THREAD

**Discussion by techcatgato** [pybricks/support#1785] [Q&A] [answered]
Hey, I know it is possible to use socket() with EV3 but is it somehow possible with spike prime too? I want to establish a connection between the hub and pc that way so the pc can use regular python.

**Comment by dlech** [comment]
On SPIKE Prime we currently only support Bluetooth Low Energy. So RFCOMM socket isn't possible.

But we recently did add a new `AppData` class that can be used to communicate to a PC if you want to help test that out. I don't think it hasn't really been documented or included in any official releases so it could be a bit of work to figure out how to use it at this point though.

**Reply by techcatgato** [reply]
So there is no place to learn about this right now but it exists right?

**Reply by dlech** [reply, marked as answer]
This is the pull request where the feature was added: https://github.com/pybricks/pybricks-micropython/pull/246 with an example hub program in https://github.com/pybricks/pybricks-micropython/pull/246#issuecomment-2243675758.

It doesn't look like we've added anything to pybricksdev to show how the PC side of things would work yet though.

**Reply by techcatgato** [reply]
How can it be received by the pc in a regular python script?

**Comment by laurensvalk** [comment]
Here is a tutorial to communicate between the hub and a computer running Python, via Bluetooth: https://pybricks.com/projects/tutorials/wireless/hub-to-device/pc-communication/

**Reply by techcatgato** [reply]
Can this be used while the robot is running normal code?

**Reply by dlech** [reply]
yes
